src/attributes.cpp

```diff
@@ -3279,7 +3279,7 @@ namespace {
 
             // copy the source file to the build dir
             Rcpp::Function filecopy = Rcpp::Environment::base_env()["file.copy"];
-            filecopy(cppSourcePath_, generatedCppSourcePath(), true);
+            filecopy(cppSourcePath_, generatedCppSourcePath(), true, Rcpp::_["copy.mode"] = false);
```

Comment:
Nice. I'll look some more at that tomorrow over morning coffee -- this ensures we get 0644 irrespective of original permissionss?

Reply:
Have by now caught up onto `help(file.copy)` -- this is perfect. Had worried that this code, because it dates back to before we had filesystem semantics from C++17 and all that, would have to do dance to do this portably.  But farming out to a call to then underlying R instance we always have is golden, and that function has that option.  Very nice.

Reply:
I generally have a mild dislike of  `Rcpp::_[...]` over the preferred `Rcpp::Named("...")` but they are equivalent, and the former is used in the file already.  So may change that another time....